### DISCUSSION TITLE
Manhattan links overlap?

### DISCUSSION THREAD

**Discussion by sabrael** [clientIO/joint#1866] [Q&A]
Working on a project to display graphical representations of our database dynamically, so we can avoid having to update them manually in the future. For some tables (those where only the top node has multiple connections), everything looks great. For others... not so much. Here's our current layout, which I'm trying to approximate with JointJS:

![desired_output](https://user-images.githubusercontent.com/32584035/198352789-d80259c8-f417-4859-87c7-d4c5d73b4eb7.png)

And here's the output I'm getting with Manhattan routing - it's stretched horizontally, which for more complicate tables could easily mean it runs off the screen. Some of the links also go right through other elements.

![current_output](https://user-images.githubusercontent.com/32584035/198353492-7b739683-665b-4a82-8f55-5414bf6b24f6.png)

If I try setting the start and end direction params on the router to make things more vertical, it gets much worse, as all of the links overlap each other:

![current_output2](https://user-images.githubusercontent.com/32584035/198353873-92616095-375b-48e2-b52f-649646cc8051.png)

Any suggestions for getting a more compact layout like our current design, and avoiding having the links overlap? Is there maybe a way to set a max number of nodes per row?

**Comment by kumilingus** [comment]
The `manhattan` router does not prevent links from overlapping each other (it only prevent links overlapping elements).

You might need to use some automatic layout for this ([DirectedGraph](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#layout.DirectedGraph), [ELK](https://resources.jointjs.com/demos/elk)).

There is an open issue in ELK repository regarding [standalone edge routing](https://github.com/eclipse/elk/issues/315).
